engines/interp_seq_mc.cpp

```diff
@@ -0,0 +1,263 @@
+/*********************                                                  */
+/*! \file interp_seq_mc.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Po-Chun Chien
+ ** This file is part of the pono project.
+ ** Copyright (c) 2019 by the authors listed in the file AUTHORS
+ ** (in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file LICENSE in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation of interpolation-sequence based model checking.
+ **
+ ** This unbounded model checking algorithm was introduced by Yakir Vizel and
+ ** Orna Grumberg in FMCAD 2009 (https://doi.org/10.1109/FMCAD.2009.5351148).
+ **
+ **/
+
+#include "engines/interp_seq_mc.h"
+
+#include "smt-switch/exceptions.h"
+#include "smt-switch/utils.h"
+#include "smt/available_solvers.h"
+#include "utils/logger.h"
+#include "utils/term_analysis.h"
+
+using namespace smt;
+
+namespace pono {
+
+InterpSeqMC::InterpSeqMC(const SafetyProperty & p,
+                         const TransitionSystem & ts,
+                         const SmtSolver & slv,
+                         PonoOptions opt)
+    : super(p, ts, slv, opt),
+      interpolator_(
+          create_interpolating_solver_for(opt.smt_interpolator_, Engine::ISMC)),
+      to_interpolator_(interpolator_),
+      to_solver_(solver_)
+{
+  engine_ = Engine::ISMC;
+}
+
+InterpSeqMC::~InterpSeqMC() {}
+
+void InterpSeqMC::initialize()
+{
+  if (initialized_) {
+    return;
+  }
+
+  super::initialize();
+
+  interpolator_->reset_assertions();
+
+  // register UF symbols
```

Comment:
minor: you meant uninterpreted symbols

Reply:
UF should be correct here.
These symbols were introduced during abstraction (e.g., ceg_proph or ceg_bv_ops).
I expanded the code comments a bit.